src/tuning.ts

```diff
@@ -19,59 +24,63 @@ export class Tuning {
 
   constructor(tuningString = 'E/5,B/4,G/4,D/4,A/3,E/3,B/2,E/2') {
     // Default to standard tuning.
+    this.tuningString = '';
+    this.tuningValues = [];
+    this.numStrings = 0;
     this.setTuning(tuningString);
   }
 
-  noteToInteger(noteString) {
+  static noteToInteger(noteString: string): number {
     return Flow.keyProperties(noteString).int_value;
   }
 
-  setTuning(noteString) {
+  setTuning(noteString: string): void {
+    let noteName = noteString;
     if (Tuning.names[noteString]) {
-      noteString = Tuning.names[noteString];
+      noteName = Tuning.names[noteString];
     }
 
-    this.tuningString = noteString;
+    this.tuningString = noteName;
     this.tuningValues = [];
     this.numStrings = 0;
 
-    const keys = noteString.split(/\s*,\s*/);
+    const keys = noteName.split(/\s*,\s*/);
     if (keys.length === 0) {
-      throw new Vex.RERR('BadArguments', 'Invalid tuning string: ' + noteString);
+      throw new Vex.RERR('BadArguments', `Invalid tuning string: ${noteString}`);
     }
 
     this.numStrings = keys.length;
-    for (let i = 0; i < this.numStrings; ++i) {
-      this.tuningValues[i] = this.noteToInteger(keys[i]);
+    for (let i = 0; i < this.numStrings; i += 1) {
+      this.tuningValues[i] = Tuning.noteToInteger(keys[i]);
     }
   }
 
-  getValueForString(stringNum) {
-    const s = parseInt(stringNum, 10);
+  getValueForString(stringNum: string | number): number {
+    const s = typeof stringNum === 'string' ? parseInt(stringNum, 10) : stringNum;
     if (s < 1 || s > this.numStrings) {
       throw new Vex.RERR('BadArguments', `String number must be between 1 and ${this.numStrings}:${stringNum}`);
     }
 
     return this.tuningValues[s - 1];
   }
 
-  getValueForFret(fretNum, stringNum) {
+  getValueForFret(fretNum: string | number, stringNum: string | number): number {
     const stringValue = this.getValueForString(stringNum);
-    const f = parseInt(fretNum, 10);
+    const f = typeof fretNum === 'string' ? parseInt(fretNum, 10) : fretNum;
 
     if (f < 0) {
-      throw new Vex.RERR('BadArguments', 'Fret number must be 0 or higher: ' + fretNum);
+      throw new Vex.RERR('BadArguments', `Fret number must be 0 or higher: ${fretNum}`);
     }
 
     return stringValue + f;
   }
 
-  getNoteForFret(fretNum, stringNum) {
+  getNoteForFret(fretNum: string | number, stringNum: string | number): string {
     const noteValue = this.getValueForFret(fretNum, stringNum);
 
     const octave = Math.floor(noteValue / 12);
     const value = noteValue % 12;
 
-    return Flow.integerToNote(value) + '/' + octave;
+    return `${Flow.integerToNote(value)}/${octave}`;
```

Comment:
Here's where I much prefer the clarity of `+` and strings -- we add six extra `${}` control characters to get rid of the space-separated elegance on seeing exactly what is concatenated, what is a function call, and what is not.  Even `return Flow.integerToNote(value) + '/' + octave.toString();` seems clearer to me

+1 on disabling this lint rule.

Reply:
This one is a style topic. In the music.ts I believe that @0xfe said that he was fine with it.

Personally I like it, it is compacter and I do not like the `+` on cases with multiple concatenations.
Setting the rule also ensures that only templates will be used and not a mix of both approaches.

@0xfe how would you like to proceed?

Reply:
Although I see what Michael is saying for this instance, I prefer this approach overall, because templates are easier to read, and `+` gets really messy and can mask bugs. So with the intent of keeping the style consistent (within reason), let's leave the template here.

Reply:
sounds good.  Will keep that in mind.